packages/cardpay-sdk/sdk/utils/general-utils.ts

```diff
@@ -381,3 +387,26 @@ export async function wait(ms = 1000) {
     setTimeout(resolve, ms);
   });
 }
+
+// Returns the current standard gas price in wei in the specified gas token
+export async function currentGasPrice(network: string, gasTokenAddress: string) {
+  let gasStationResponse = await fetch(`${getConstantByNetwork('relayServiceURL', network)}/v1/gas-station/`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+  });
+
+  let gasPriceInNativeToken = (await gasStationResponse.json()).standard;
+
+  let exchangeAddress = getAddressByNetwork('scheduledPaymentExchange', network);
+  let wrappedNativeTokenAddress = getAddressByNetwork('wrappedNativeToken', network);
+  const provider = new JsonRpcProvider(getConstantByNetwork('rpcUrl', network));
+  let contract = new EthersContract(exchangeAddress, ExchangeABI, provider);
+
+  let gasTokenInUsd = (await contract.exchangeRateOf(gasTokenAddress)).value;
```

Comment:
If `gasTokenAddress == usdTokenInExchangeContract` we don't need to call the `exchangeRateFunction`, define `gasTokenInUsd = 1e18`

Reply:
we should be using BigNumber (or BigInteger) though right? don't use a native javascript number for any of this math.

Reply:
right